### PR TITLE
Use pmemfile in travis tests

### PR DESCRIPTION
**Pull request by GBuella**
This uses cmake options not in pmemfile yet, see:
https://github.com/pmem/pmemfile/pull/339

This change is [Reviewable](https://reviewable.io/reviews/pmem/syscall_intercept/48)

### CODE REVIEW

**Comment by krzycz** [comment]
Reviewed 5 of 6 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[utils/docker/images/build-pmemfile.sh, line 37 at r2](https://reviewable.io:443/reviews/pmem/syscall_intercept/48#-Ktjm960uZ-aZR_4FAvS:-Ktjm9611zVJ00S_U-KK:b-h1durq) ([raw file](https://github.com/pmem/syscall_intercept/blob/734ad088557a1f577099b6c8fcaaf20cd9cbeecf/utils/docker/images/build-pmemfile.sh#L37)):*
> ```Shell
> #
> 
> # libpmemfile needs syscall_intercept to build, so create a mock library
> ```

Is it a problem to clone/build the real syscall_intercept or include it in the docker image?

---

*[utils/docker/images/Dockerfile.ubuntu-16.04, line 76 at r1](https://reviewable.io:443/reviews/pmem/syscall_intercept/48#-Kth4gQQ5kobQlDSEHWZ:-Kth4gQR4so7DvMyI6RS:bm57acp) ([raw file](https://github.com/pmem/syscall_intercept/blob/f00b379045ce8015c84762db6ccb5aa551bcac77/utils/docker/images/Dockerfile.ubuntu-16.04#L76)):*
> ```
> 
> # Install nvml
> COPY install-nvml.sh install-nvml.sh
> ```

That's weird...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/syscall_intercept/48)*

**Comment by GBuella** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[utils/docker/images/build-pmemfile.sh, line 37 at r2](https://reviewable.io:443/reviews/pmem/syscall_intercept/48#-Ktjm960uZ-aZR_4FAvS:-KtkHoEroK5q9CI2HsJf:b-p8iysq) ([raw file](https://github.com/pmem/syscall_intercept/blob/734ad088557a1f577099b6c8fcaaf20cd9cbeecf/utils/docker/images/build-pmemfile.sh#L37)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

Is it a problem to clone/build the real syscall_intercept or include it in the docker image?
</blockquote></details>

That would mean doing a lot of things that are not needed, which is a problem in my eyes.

---

*[utils/docker/images/Dockerfile.ubuntu-16.04, line 76 at r1](https://reviewable.io:443/reviews/pmem/syscall_intercept/48#-Kth4gQQ5kobQlDSEHWZ:-KtkINrsKxE30TCLdv9j:b3sr9ao) ([raw file](https://github.com/pmem/syscall_intercept/blob/f00b379045ce8015c84762db6ccb5aa551bcac77/utils/docker/images/Dockerfile.ubuntu-16.04#L76)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

That's weird...
</blockquote></details>

Install NVML!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/syscall_intercept/48)*

**Comment by GBuella** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[utils/docker/images/Dockerfile.ubuntu-16.04, line 76 at r1](https://reviewable.io:443/reviews/pmem/syscall_intercept/48#-Kth4gQQ5kobQlDSEHWZ:-KtkIW-SxEcV9LOY_U8D:bi6dgfg) ([raw file](https://github.com/pmem/syscall_intercept/blob/f00b379045ce8015c84762db6ccb5aa551bcac77/utils/docker/images/Dockerfile.ubuntu-16.04#L76)):*
<details><summary><i>Previously, GBuella (Gabor Buella) wrote…</i></summary><blockquote>

Install NVML!
</blockquote></details>

This line just install the install_nvml to install_nvml, in order to install NVML

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/syscall_intercept/48)*

**Comment by krzycz** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[utils/docker/images/Dockerfile.ubuntu-16.04, line 76 at r1](https://reviewable.io:443/reviews/pmem/syscall_intercept/48#-Kth4gQQ5kobQlDSEHWZ:-KtkN-BIr10Z33fy8nxt:b-1yrzes) ([raw file](https://github.com/pmem/syscall_intercept/blob/f00b379045ce8015c84762db6ccb5aa551bcac77/utils/docker/images/Dockerfile.ubuntu-16.04#L76)):*
<details><summary><i>Previously, GBuella (Gabor Buella) wrote…</i></summary><blockquote>

This line just install the install_nvml to install_nvml, in order to install NVML
</blockquote></details>

Well, I meant the syntax is weird, as it suggests you copy a file to the same file/location, which is pointless. When looking at this you have no idea where is the source and destination.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/syscall_intercept/48)*

**Comment by GBuella** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[utils/docker/images/Dockerfile.ubuntu-16.04, line 76 at r1](https://reviewable.io:443/reviews/pmem/syscall_intercept/48#-Kth4gQQ5kobQlDSEHWZ:-KtkNi8PJ2myWrMuM8hC:bjnm6m8) ([raw file](https://github.com/pmem/syscall_intercept/blob/f00b379045ce8015c84762db6ccb5aa551bcac77/utils/docker/images/Dockerfile.ubuntu-16.04#L76)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

Well, I meant the syntax is weird, as it suggests you copy a file to the same file/location, which is pointless. When looking at this you have no idea where is the source and destination.
</blockquote></details>

Exactly what I was thinking the first time I'v seen this dockerfile syntax, but then I figured it out quickly.
Do you have any suggestion to improve the situation?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/syscall_intercept/48)*

**Comment by krzycz** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[utils/docker/images/Dockerfile.ubuntu-16.04, line 76 at r1](https://reviewable.io:443/reviews/pmem/syscall_intercept/48#-Kth4gQQ5kobQlDSEHWZ:-KtkSP21Syd9h19FJP2l:bx3l7c0) ([raw file](https://github.com/pmem/syscall_intercept/blob/f00b379045ce8015c84762db6ccb5aa551bcac77/utils/docker/images/Dockerfile.ubuntu-16.04#L76)):*
<details><summary><i>Previously, GBuella (Gabor Buella) wrote…</i></summary><blockquote>

Exactly what I was thinking the first time I'v seen this dockerfile syntax, but then I figured it out quickly.
Do you have any suggestion to improve the situation?
</blockquote></details>

No, that's perfectly fine.  I just expressed my opinion on dockerfile syntax. ;-)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/syscall_intercept/48)*

**Comment by krzycz** [comment]
:lgtm:

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/syscall_intercept/48#-:-KtkSz_f8NU1Ic10vDqj:bnfp4nl)*

File: utils/docker/images/build-pmemfile.sh [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+mkdir build
+cd build
+cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+	-DBUILD_LIBPMEMFILE=ON \
+	-DAUTO_GENERATE_SOURCES=OFF \
+	-DTESTS_USE_FORCED_PMEM=ON \
+	-DDEVELOPER_MODE=OFF \
+	-DFAULT_INJECTION=OFF \
+	-DLONG_TESTS=OFF \
+	-DTEST_PROCESS_SWITCHING=OFF \
```
Comment (marcinslusarz) [inline utils/docker/images/build-pmemfile.sh]: This doesn't exist anymore.

File: utils/docker/images/build-pmemfile.sh [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	-DTRACE_TESTS=ON \
+	-DUSE_ASAN=OFF \
+	-DUSE_UBSAN=OFF \
+	-DANTOOL_TESTS=SKIP \
+	-DBUILD_LIBPMEMFILE_POSIX_TESTS=OFF \
+	-DBUILD_LIBPMEMFILE_TESTS=ON \
```
Comment (marcinslusarz) [inline utils/docker/images/build-pmemfile.sh]: These 2 options don't exist yet.

File: utils/docker/images/build-pmemfile.sh (hunk 1 above)
Comment (GBuella) [reply utils/docker/images/build-pmemfile.sh]: I anticipate the past apparently.

File: utils/docker/images/build-pmemfile.sh (hunk 2 above)
Comment (GBuella) [reply utils/docker/images/build-pmemfile.sh]: I anticipate the future!

File: utils/docker/images/build-pmemfile.sh (hunk 1 above)
Comment (GBuella) [reply utils/docker/images/build-pmemfile.sh]: And what are these about?
https://github.com/pmem/pmemfile/blob/master/tests/preload/preload-helpers.cmake#L37
https://github.com/pmem/pmemfile/blob/master/tests/preload/CMakeLists.txt#L114
https://github.com/pmem/pmemfile/blob/master/tests/preload/pool_locking/pool_locking.cmake#L43

Comment (marcinslusarz) [reply utils/docker/images/build-pmemfile.sh]: Internal uses.

**Comment by marcinslusarz** [comment]
:lgtm:

---

Reviewed 5 of 6 files at r1, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/syscall_intercept/48#-:-Ktkjd4TWAQ_Qb7v5NQV:bnfp4nl)*

File: utils/docker/images/build-pmemfile.sh (hunk 2 above)
Comment (GBuella) [reply utils/docker/images/build-pmemfile.sh]: And future is now!

**Pull request merged**